### PR TITLE
New Badge Component

### PR DESCRIPTION
**Pull request by KushalNerella07**
Implemented Badge component:

- Introduced `size` prop to adjust the badge size (xsmall, small, medium, large).
- Introduced `shape` prop to specify the shape of the badge (circle, square, rectangle).
- Allowed for custom colors through the `color` prop.
- Improved Storybook controls to showcase both string and number types for the `content` prop.

### CODE REVIEW

File: src/packages/Badge/styles.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import styled from 'styled-components'
+import { BadgeProps } from '.'
+
+export const Badge = styled.div<Pick<BadgeProps, 'size' | 'shape' | 'color'>>`
+  display: inline-block;
+  border-radius: ${({ shape }) =>
+    shape === 'circle' ? '50%' : shape === 'rectangle' ? '0px' : '0px'};
+  font-size: ${({ size }) =>
+    size === 'xsmall'
```
Comment (vczb) [inline src/packages/Badge/styles.tsx:9]: hey let's try to set the sizes using theme

here is a reference to follow

https://github.com/vczb/sagu-ui/blob/fba435fb6b57daa93d7ab596d73be30e68a6393e/src/packages/NavLink/styles.ts#L12

if it doesn't work for you just pick the size from the theme
https://github.com/vczb/sagu-ui/blob/fba435fb6b57daa93d7ab596d73be30e68a6393e/src/styles/theme/index.ts#L37

File: src/packages/Badge/styles.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      ? '0.6em'
+      : size === 'small'
+      ? '0.7em'
```
Comment (vczb) [inline src/packages/Badge/styles.tsx:12]: this is the sizing scale that we currently use, pick from the scale

```
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      huge: '5.2rem'
    },
  spacings: {
    none: '0',
    xmini: '0.2rem',
    mini: '0.4rem',
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  ```

File: src/packages/Badge/styles.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import styled from 'styled-components'
+import { BadgeProps } from '.'
+
+export const Badge = styled.div<Pick<BadgeProps, 'size' | 'shape' | 'color'>>`
+  display: inline-block;
+  border-radius: ${({ shape }) =>
+    shape === 'circle' ? '50%' : shape === 'rectangle' ? '0px' : '0px'};
```
Comment (vczb) [inline src/packages/Badge/styles.tsx:7]: you can add those borders to theme and use them
https://github.com/vczb/sagu-ui/blob/fba435fb6b57daa93d7ab596d73be30e68a6393e/src/styles/theme/index.ts#L10

for the 50% i suggest one of those names "rounded" or "circle"

for 0 you can use none, ps: dont need px when you set as 0

**Review by vczb — COMMENTED** [review]
Thanks for your PR, I left some comments, let me know if you have any questions

**Review by vczb — APPROVED** [review]
LGTM

**Comment by KushalNerella07** [comment]
I can make more components if you want me to. Please make a request and I can get working on them

**Comment by KushalNerella07** [comment]
I just realized I can make one too. @vczb would you be open for more components?

**Comment by vczb** [comment]
> I just realized I can make one too. @vczb would you be open for more components?

Yes, you can submit as many as you want to :) [please double-check to avoid creating a duplicated ones](https://github.com/vczb/sagu-ui/tree/main/src/packages)

**Pull request merged**